cmd/gh/main.go

```diff
@@ -45,6 +45,12 @@ func main() {
 			fmt.Fprintf(stderr, "failed to process aliases:  %s\n", err)
 			os.Exit(2)
 		}
+
+		if expandedArgs == nil && err == nil {
+			// It was an external alias; we ran it and are now done.
+			os.Exit(0)
```

Comment:
The added comment helped me understand why we exit here when both values are nil; thank you.

I'm feeling uneasy about a function called `ExpandAlias` also executing an external process if the alias starts with `!`, and otherwise just returning a slice of strings. I'm thinking that the function is too overloaded, since it has very different responsibilities depending on whether an alias starts with `!`. Would you be open to changing `ExpandAlias` to always return a slice of strings and a boolean value that dictates whether those arguments should be run as an external process or through Cobra?

With that, the `main.go` implementation will be in charge of actually spawning that external process, which I feel is a more suitable place for it since it's also in charge of dispatching Cobra arguments and handling errors.

Reply:
That's fair. I think I was just trying to keep all of my changes in one small box but `ExpandAlias` is definitely too monstrous now.